tests/acceptance/watch-query-test.js

```diff
@@ -0,0 +1,57 @@
+import { module, test } from 'qunit';
+import { setupApplicationTest } from "dummy/tests/helpers/setup";
+import { addResolveFunctionsToSchema } from 'graphql-tools';
+import { click, currentURL, find, visit } from "@ember/test-helpers";
+
+const mockHuman = {
+  id: '1000',
+  name: 'Luke Skywalker',
+  __typename: 'Human',
+};
+
+module('Acceptance | watch query', function(hooks) {
+  setupApplicationTest(hooks);
+
+  let schema;
+
+  hooks.beforeEach(function() {
+    schema = this.pretender.schema;
+  });
+
+  test('visiting /luke', async function(assert) {
+    let done = assert.async();
+    let character = Object.assign({}, mockHuman);
+    let resolvers = {
+      Query: {
+        human(obj, args) {
+          assert.deepEqual(args, { id: '1000' });
+          return Object.assign({}, character);
+        },
+      },
+      Mutation: {
+        changeCharacterName(_, { id, name }) {
+          assert.equal(id, mockHuman.id, 'correct mutuation id is passed in');
+          assert.equal(name, 'Han Solo', 'correct mutuation id is passed in');
+          return Object.assign(character, { name });
```

Comment:
On line 28 above, you're making a copy of `character` into a new `Object`, but here you're mutating `character`. I'd be less worried about side effects if this was instead:

```js
return Object.assign({}, character, { name });
```